nodi_backend/apps/datasets/views.py

```diff
@@ -129,18 +129,34 @@ def search(self, request):
         datatypes = request.query_params.get('datatypes', [])
         categories = request.query_params.get('categories', [])
 
+        #filters based on name query
+        #-takes out categories mentioned in name and filters full set using them
+        #-if categories query exist:
+        #   -adds categories-in-name to categories query so they don't get filtered out later
+        #-filters full set using the words-in-name
+        #-unions the two filtered sets
         if name:
-            wordsinname = name.split(" ")
-            for word in wordsinname:
+            sSet = DataSet.objects.none()
+            cats = ["Finances", "Safety", "Student Life", "Admissions", "Academic", "Campus"]
```

Comment:
Don't make static variables. Use what exists in the db.
`cats = Category.objects.all()` or something like it.